noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/components/public_kernel_output_composer.nr

```diff
@@ -31,39 +31,36 @@ impl PublicKernelOutputComposer {
             let _ = self.output_builder.end.public_call_stack.pop();
         }
         if phase == PublicKernelPhase.TEARDOWN {
-            let _ = self.output_builder.public_teardown_call_stack.pop();
+            self.output_builder.public_teardown_call_request = PublicCallRequest::empty();
         }
         *self
     }
 
-    pub fn propagate_from_public_call(&mut self, public_call: PublicCircuitPublicInputs, phase: u8) -> Self {
-        self.output_builder.constants.global_variables = public_call.global_variables;
-        self.propagate_revert_code(public_call, phase);
-        self.propagate_validation_requests(public_call);
-        self.propagate_accumulated_data(public_call, phase);
-        self.update_gas_used(public_call, phase);
+    pub fn propagate_from_enqueued_call(&mut self, enqueued_call: VMCircuitPublicInputs, phase: u8) -> Self {
+        self.output_builder.constants.global_variables = enqueued_call.constants.global_variables;
+        self.propagate_revert_code(enqueued_call, phase);
+        self.propagate_validation_requests(enqueued_call);
+        self.propagate_accumulated_data(enqueued_call, phase);
+        self.propagate_end_side_effect_counter(enqueued_call);
+        self.update_gas_used(enqueued_call, phase);
         *self
     }
 
     pub fn finish(self) -> PublicKernelCircuitPublicInputs {
         self.output_builder.finish()
     }
 
-    fn propagate_revert_code(&mut self, public_call: PublicCircuitPublicInputs, phase: u8) {
-        let prev_revert_code = self.output_builder.revert_code;
-        let public_call_revert_code = public_call.revert_code;
-        if phase == PublicKernelPhase.SETUP {
-            // Keep prev_revert_code.
-            // public_call_revert_code should be 0. If not, PublicCallDataValidator > validate_revert_code will fail.
-        }
-        if phase == PublicKernelPhase.APP_LOGIC {
-            if prev_revert_code == 0 {
-                self.output_builder.revert_code = public_call_revert_code;
+    fn propagate_revert_code(&mut self, enqueued_call: VMCircuitPublicInputs, phase: u8) {
+        if enqueued_call.reverted {
+            let prev_revert_code = self.output_builder.revert_code;
+            if phase == PublicKernelPhase.SETUP {
+                // enqueued_call.reverted should be false. If not, EnqueuedCallDataValidator > validate_revert_flag will fail.
             }
```

Comment:
Actually, I think I see that asserted in `validate_revert_flag`

Reply:
Yes, it will fail in EnqueuedCallDataValidator. And once the enqueued call is validated, the flag in the output must be 0, hence the assert_eq(0) here for SETUP.